X16 Reference - Appendix F - 65C816 Processor.md

```diff
@@ -0,0 +1,2470 @@
+RAM.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TAX
+
+**Transfer Accumulator to Index X**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+TAX            imp           AA  1   2           n.....z. .
+```
+
+Copies the contents of .A to .X.
+
+The transfer (copy) is 16 bits only if **m** and **x** are both 0.
+
+If **m** or **x** is 1, the copy will only be 8 bits, and the upper byte of .X
+will nt be affected.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TAY
+
+**Transfer Accumulator to Index X**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+TAY            imp           A8  1   2           n.....z. .
+```
+
+Copies the contents of .A to .Y.
+
+The transfer (copy) is 16 bits only if **m** and **x** are both 0.
+
+If **m** or **x** is 1, the copy will only be 8 bits, and the upper byte of .Y
+will nt be affected.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TCD
+
+**Transfer C Accumulator to Direct Register**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+TCD            imp           5B  1   2           n.....z. .
+```
+
+This is one of the times that the 16-bit Accumulator is called .C. This copies
+the 16-bit value from the 16-bit Accumulator to the Stack Pointer
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TCS
+
+**Transfer C Accumulator to Stack Pointer**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+TCS            imp           1B  1   2           ........ .
+```
+
+This is one of the times that the 16-bit Accumulator is called .C. This copies
+the 16-bit value from the 16-bit Accumulator to the Stack Pointer
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TDC
+
+**Transfer Direct Register to C Accumulator**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+TDC            imp           7B  1   2           n.....z. .
+```
+
+Copies the value of the Direct Register to the Accumulator.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TRB
+
+**Test and Reset Bit**
+
+```text
+SYNTAX         MODE          HEX LEN CYCLES      FLAGS   
+TRB $20        dir           14  2   7-2*m+w     ......z. .
+TRB $1234      abs           1C  3   8-2*m       ......z. .
+```
+
+Performs a bitwise AND with a memory value and the Accumulator. When the result
+of the AND operation is Zero, **z** is set.
+
+After this, the bits that were 1 in the Accumulator are cleared.
```

Comment:
This wording is confusing. The accumulator does not change, only memory.

Reply:
New description:

TRB clears bits in memory, based on the bits that were set in the accumulator.
This can be used to selectively clear or reset specific bits in a memory value.

TRB also performs a bitwise AND with a memory value and the Accumulator. When
the result of the AND operation is Zero, **z** is set.